pkg/kubelet/kubelet.go

```diff
@@ -745,7 +747,19 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	// check node capabilities since the mount path is not the default
 	if len(kubeCfg.ExperimentalMounterPath) != 0 {
 		kubeCfg.ExperimentalCheckNodeCapabilitiesBeforeMount = false
+
+		if len(klet.clusterDNS) > 0 {
+			resolvePath := filepath.Join(strings.TrimSuffix(kubeCfg.ExperimentalMounterPath, "/mounter"), "rootfs", "etc", "resolv.conf")
+			dnsString := ""
+			for _, dns := range klet.clusterDNS {
+				dnsString = dnsString + fmt.Sprintf("nameserver %s\n", dns)
+			}
+
+			ioutil.WriteFile(resolvePath, []byte(dnsString), 0600)
```

Comment:
In this way the mounter wouldn't be able to utilize the internal search paths (e.g. `"c.[PROJECT_ID].internal.", "google.internal."`), but I guess that is okay?

Ref: https://cloud.google.com/compute/docs/vpc/internal-dns

Reply:
Oh and we should check error here?

Reply:
What does it mean to not utilize the internal search paths?  The NFS or Gluster service could be setup directly on a GCE instance instead of as a Pod.

Reply:
It means user will have to specify FQDN instead of relying on resolver automatically appending those search paths.

> The NFS or Gluster service could be setup directly on a GCE instance instead of as a Pod.

Humm, then seems like this will change the behavior (e.g. `INSTANCE_NAME` alone is not resolvable anymore) and may break existing users?

Reply:
If a user is migrating from debian to gci, then it will break.

In GCI, we had this timeline:
Initial: DNS resolution failed to work at all
1.6.?, 1.7.? patch: DNS resolution works for non-Pod based services.

Reply:
An alternative is to rewrite the nameserver only, while keeping the original search paths in resolv.conf. There are some similar logic in GetClusterDNS():
https://github.com/kubernetes/kubernetes/blob/7c70decd278f574d704ba16aab9e9f4d906549ef/pkg/kubelet/kubelet.go#L1341-L1344

Reply:
ok, I also write the search string as suggested. So this time non-pod nfs service will also work, right?